File: day-1/adventofcode-2021-day-1-swift/adventofcode-2021-day-1/MeasurementWindow.swift
//
//  MeasurementWindow.swift
//  adventofcode-2021-day-1
//
//  Created by Sören Kuklau on 01.12.21.
//

import Foundation

class MeasurementWindow {
    var Values : [Int] = []

    func addValue(newValue : Int) {
        while Values.count >= 3 {
            Values.remove(at: 0)
        }
        
        Values.append(newValue)
    }
}

File: day-1/adventofcode-2021-day-1-csharp/adventofcode-2021-day-1/MeasurementWindow.cs
namespace adventofcode_2021_day_1
{
    public class MeasurementWindow
    {
        public List<int> Values { get; } = new();

        public void AddValue(int newValue)
        {
            while (Values.Count >= 3)
                Values.RemoveAt(0);

            Values.Add(newValue);
        }
    }
}

